plugins/ta/plugin_ta.c

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Open Mobile Platform LLC
+ */
+
+#include <assert.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+/* This TA header */
+#include <plugin_ta.h>
+
+/* Interface with optee-os pta to connect to any plugins */
+#include <pta_supp_plugin.h>
+
+#include <string.h>
+#include <stdint.h>
+
+/*
+ * Interface with syslog tee-supplicant plugin
+ */
+#define SYSLOG_PLUGIN_UUID { 0x96bcf744, 0x4f72, 0x4866, \
+		{ 0xbf, 0x1d, 0x86, 0x34, 0xfd, 0x9c, 0x65, 0xe5 } }
+#define TO_SYSLOG_CMD 0
+
+/* according to syslog.h */
+#define LOG_EMERG 0 /* system is unusable */
+#define LOG_ALERT 1 /* action must be taken immediately */
+#define LOG_CRIT 2 /* critical conditions */
+#define LOG_ERR 3 /* error conditions */
+#define LOG_WARNING 4 /* warning conditions */
+#define LOG_NOTICE 5 /* normal but significant condition */
+#define LOG_INFO 6 /* informational */
+#define LOG_DEBUG 7 /* debug-level messages */
+
+/*
+ * Called when the instance of the TA is created. This is the first call in
+ * the TA.
+ */
+TEE_Result TA_CreateEntryPoint(void)
+{
+	return TEE_SUCCESS;
+}
+
+/*
+ * Called when the instance of the TA is destroyed if the TA has not
+ * crashed or panicked. This is the last call in the TA.
+ */
+void TA_DestroyEntryPoint(void)
+{
+}
+
+/*
+ * Called when a new session is opened to the TA. *sess_ctx can be updated
+ * with a value to be able to identify this session in subsequent calls to the
+ * TA. In this function you will normally do the global initialization for the
+ * TA.
+ */
+TEE_Result TA_OpenSessionEntryPoint(uint32_t param_types,
+				    TEE_Param __maybe_unused params[4],
+				    void __maybe_unused **sess_ctx)
+{
+	uint32_t exp_param_types =
+		TEE_PARAM_TYPES(TEE_PARAM_TYPE_NONE, TEE_PARAM_TYPE_NONE,
+				TEE_PARAM_TYPE_NONE, TEE_PARAM_TYPE_NONE);
+
+	if (param_types != exp_param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/* If return value != TEE_SUCCESS the session will not be created. */
+	return TEE_SUCCESS;
+}
+
+/*
+ * Called when a session is closed, sess_ctx hold the value that was
+ * assigned by TA_OpenSessionEntryPoint().
+ */
+void TA_CloseSessionEntryPoint(void __unused *sess_ctx)
+{
+}
+
+static TEE_Result syslog_plugin_ping(void)
+{
+	TEE_Result res;
```

Comment:
Prefer to follow the same codiing rules as optee_os/optee_client: init all local vars.